infer/environment.cc

```diff
@@ -791,8 +791,10 @@ core::TypePtr Environment::processBinding(core::Context ctx, cfg::Binding &bind,
                     retainedResult = make_shared<core::DispatchResult>(std::move(dispatched));
                 }
                 if (lspQueryMatch) {
```

Comment:
Nit: I think this would be easier to read with...

```c++
if (lspQueryMatch && retainedResult->main.method.exists() && !retainedResult.main.method.isSynthetic()) {
 // push
}
```

Reply:
Yup, I can do that.

Reply:
As per offline conversation, it doesn't fix #1777, but I added a test for that to make it slightly easier for the next person to touch this.

Reply:
This was originally:

   if (a) {
    if (!(b && c)) {
      // push

After I applied the suggestion to make it:

  if (a && b && !c) {
    // push

I realized that tests started to fail because this isn't the same logic for the case where a=true and b=false.   So, I just pushed a change that attempted to capture the spirit of your suggestion by combining the line into one expression:

    if (a && !(b && c))

Although, due to line-length, this autoformats to:

    if (a &&
        !(b && c))
      // push

Which looks a lot like the original 🙄 , but just so you know it's not for lack of trying.   🤷‍♂

Reply:
I knew they were different. Now we'll push sends where the method *doesn't* exist but is not synthetic, which I *thought* would be good behavior. 

However, autocomplete *does* want non-existent methods because the target of an autocomplete might not exist!

e.g. typing `fo` which autocompletes to `foo`, Sorbet creates the non-existent method `fo` which autocomplete finds by location and uses to recommend `foo`.

Good catch.